assets/prototype/domain/eventEditor/data/queries/priceTypes/usePriceTypeForPrice.ts

```diff
@@ -1,28 +1,28 @@
 import isEmpty from 'ramda/src/isEmpty';
-import type from 'ramda/src/type';
 import useRelations from '../../../../../application/services/apollo/relations/useRelations';
 import usePriceTypes from './usePriceTypes';
 import useDefaultPriceType from './useDefaultPriceType';
+import { PriceType, EntityId } from '../../types';
 
 /**
  * A custom react hook for retrieving the related priceType from cache for the given Price entity
  *
  * @param {string} priceId price.id
  */
-const usePriceTypeForPrice = (priceId) => {
+const usePriceTypeForPrice = (priceId: EntityId): PriceType => {
 	const { getRelations } = useRelations();
 	// get related priceTypes for this price
-	const relatedPriceTypeIds = getRelations({
+	const relatedPriceTypeIds: EntityId[] = getRelations({
 		entity: 'prices',
 		entityId: priceId,
 		relation: 'priceTypes',
 	});
 
 	// get the default price type object
-	const defaultPriceType = useDefaultPriceType();
-	const relatedPriceTypes = usePriceTypes(relatedPriceTypeIds);
+	const defaultPriceType: PriceType = useDefaultPriceType();
+	const relatedPriceTypes: PriceType[] = usePriceTypes(relatedPriceTypeIds);
```

Comment:
TS is smart enough to detect some underlying logic:
![image](https://user-images.githubusercontent.com/1477453/71372619-91992180-25bd-11ea-96c2-93196584faaf.png)